src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './ContextMenu.module.css';
+import { Card } from '../Card/Сard.tsx';
+
+interface ContextMenuProps {
+	items: { label: string; action: () => void }[];
+	children: React.ReactElement<{ onContextMenu: (event: React.MouseEvent<HTMLElement>) => void }>;
+
+}
+
+export function ContextMenu({ children, items }: ContextMenuProps) {
+	const [isVisible, setIsVisible] = useState(false);
+	const [position, setPosition] = useState({ top: 0, left: 0 });
+	const menuRef = useRef<HTMLDivElement>(null);
+
+	const handleContextMenu = (e: React.MouseEvent<HTMLElement>) => {
+		e.preventDefault();
+		const { clientX, clientY } = e;
+		setPosition({ top: clientY, left: clientX });
+		setIsVisible(true);
+	};
+
+	const handleOutsideClick = (e: MouseEvent) => {
+		if (menuRef.current && !menuRef.current.contains(e.target as Node)) {
+			setIsVisible(false);
+		}
+	};
+
+	// Attach event listeners
+	useEffect(() => {
+		document.addEventListener('click', handleOutsideClick);
+		return () => {
+			document.removeEventListener('click', handleOutsideClick);
+		};
+	}, []);
+
+	return (
+		<>
+			{children && React.cloneElement(children, { onContextMenu: handleContextMenu })}
+			{isVisible && (
+				<div ref={menuRef}>
+					<Card
+						className={styles.contextMenu}
+						style={{ top: position.top, left: position.left }}
+
+					>
+						{items.map((item, index) => (
```

Comment:
JFYI: It's better to avoid using indexes as keys
https://www.developerway.com/posts/react-key-attribute#part3

Reply:
They say that: [there is no harm in using the array’s index as “key” in “static” lists - those whose items number and order stay the same](https://www.developerway.com/posts/react-key-attribute#part3:~:text=there%20is%20no%20harm%20in%20using%20the%20array%E2%80%99s%20index%20as%20%E2%80%9Ckey%E2%80%9D%20in%20%E2%80%9Cstatic%E2%80%9D%20lists%20%2D%20those%20whose%20items%20number%20and%20order%20stay%20the%20same)

Reply:
That's ok. I sent a link you get familiar with topic